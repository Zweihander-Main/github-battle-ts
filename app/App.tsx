import * as React from 'react';
import './index.css';
import { ThemeProvider } from './contexts/theme';
import Nav from './components/Nav';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Loading from './components/Loading';
import { AppState } from './types';

const Popular = React.lazy(() => import('./components/Popular'));
const Battle = React.lazy(() => import('./components/Battle'));
const Results = React.lazy(() => import('./components/Results'));

/**
 * Sets up application including theme context and routing
 *
 * @class      App (name)
 */
export default class App extends React.Component<Record<null, null>, AppState> {
	state = {
		theme: 'light',
		toggleTheme: (): void => {
			this.setState(({ theme }: AppState) => ({
				theme: theme === 'light' ? 'dark' : 'light',
			}));
		},
	} as AppState;

	render(): React.ReactNode {
		return (
			<Router>
				<ThemeProvider value={this.state}>
					<div className={this.state.theme}>
						<div className="container">
							<Nav />
							<React.Suspense fallback={<Loading />}>
								<Switch>
									<Route exact path="/" component={Popular} />
									<Route
										exact
										path="/battle"
										component={Battle}
									/>
									<Route
										path="/battle/results"
										component={Results}
									/>
									<Route
										render={(): React.ReactNode => (
											<h1>404</h1>
										)}
									/>
								</Switch>
							</React.Suspense>
						</div>
					</div>
				</ThemeProvider>
			</Router>
		);
	}
}
