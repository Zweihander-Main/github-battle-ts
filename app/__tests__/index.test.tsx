import React from 'react';
import App from '../App';
import { render, fireEvent, screen } from '@testing-library/react';

test('Popular and battle pages match snapshots', () => {
	const { asFragment } = render(<App />);
	expect(asFragment()).toMatchSnapshot();
	fireEvent.click(screen.getByText('Battle'));
	expect(asFragment()).toMatchSnapshot();
});
