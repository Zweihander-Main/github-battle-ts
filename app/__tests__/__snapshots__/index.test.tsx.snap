// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular and battle pages match snapshots 1`] = `
<DocumentFragment>
  <div
    class="light"
  >
    <div
      class="container"
    >
      <nav
        class="row space-between"
      >
        <ul
          class="row nav"
        >
          <li>
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
              style="color: rgb(187, 46, 31);"
            >
              Popular
            </a>
          </li>
          <li>
            <a
              class="nav-link"
              href="/battle"
            >
              Battle
            </a>
          </li>
        </ul>
        <button
          class="btn-clear"
          style="font-size: 30px;"
        >
          ðŸ”¦
        </button>
      </nav>
      <p
        style="font-size: 35px; position: absolute; left: 0px; right: 0px; margin-top: 20px; text-align: center;"
      >
        Loading
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popular and battle pages match snapshots 2`] = `
<DocumentFragment>
  <div
    class="light"
  >
    <div
      class="container"
    >
      <nav
        class="row space-between"
      >
        <ul
          class="row nav"
        >
          <li>
            <a
              class="nav-link"
              href="/"
              style=""
            >
              Popular
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="nav-link active"
              href="/battle"
              style="color: rgb(187, 46, 31);"
            >
              Battle
            </a>
          </li>
        </ul>
        <button
          class="btn-clear"
          style="font-size: 30px;"
        >
          ðŸ”¦
        </button>
      </nav>
      <p
        style="font-size: 35px; position: absolute; left: 0px; right: 0px; margin-top: 20px; text-align: center;"
      >
        Loading
      </p>
    </div>
  </div>
</DocumentFragment>
`;
